packages/mobile/src/services/offline-downloader/offline-storage.ts

```diff
@@ -0,0 +1,80 @@
+import path from 'path'
+
+import type { Track } from '@audius/common'
+import RNFS, { exists } from 'react-native-fs'
+
+import { getCoverArtUri } from './offline-downloader'
+
+export const downloadsRoot = path.join(RNFS.CachesDirectoryPath, 'downloads')
+
+export const getPathFromRoot = (string: string) => {
+  return string.replace(downloadsRoot, '~')
+}
+
+export const getLocalTrackRoot = (track: any) => {
+  return path.join(downloadsRoot, `/tracks`)
+}
+
+// Track Json
+
+export const getLocalTrackJsonPath = (track: Track) => {
+  return path.join(getLocalTrackRoot(track), `${track.track_id}.json`)
+}
+
+// Cover Art
+
+export const getLocalCoverArtPath = (track: Track): [string?, string?] => {
+  const uri = getCoverArtUri(track)
+  if (!uri) return []
+  const fileName = getArtFileNameFromUri(uri)
+  return [`${downloadsRoot}/tracks/${track.track_id}`, fileName]
+}
+
+export const getArtFileNameFromUri = (uri: string) => {
+  return uri.split('/').slice(-1)[0]
+}
+
+// Audio
+
+export const getLocalAudioPath = (track: Track): [string, string] => {
+  // @ts-ignore route_id exists on track
+  const fileName = `${track.route_id.replaceAll('/', '_')}.mp3`
+  return [`${downloadsRoot}/tracks/${track.track_id}`, fileName]
+}
+
+export const isAudioAvailableOffline = async (track: Track) => {
+  const fullFilePath = path.join(...getLocalAudioPath(track))
+  return await exists(fullFilePath)
+}
+
+// Storage management
+
+// Debugging method to clear all downloaded content
```

Comment:
```suggestion
/** Debugging method to clear all downloaded content */
```
super nit but nicer for code editors

Reply:
(same for func below)